lib/rails/html/sanitizer.rb

```diff
@@ -136,6 +136,10 @@ def allowed_tags(options)
       def allowed_attributes(options)
         options[:attributes] || self.class.allowed_attributes
       end
+
+      def properly_encode(fragment, options)
+        fragment.xml? ? fragment.to_xml(options) : fragment.to_html(options)
```

Comment:
I don't know how likely it is to sanitize an XML document, but I added it for completeness.